managed/src/main/java/com/yugabyte/yw/controllers/EncryptionAtRestController.java

```diff
@@ -44,13 +49,44 @@
 
 public class EncryptionAtRestController extends AuthenticatedController {
     public static final Logger LOG = LoggerFactory.getLogger(EncryptionAtRestController.class);
+    static final String SMARTKEY_KMS_PROVIDER = "SMARTKEY";
+    static final String AWS_KMS_PROVIDER = "aws";
 
     @Inject
     EncryptionAtRestManager keyManager;
 
     @Inject
     Commissioner commissioner;
 
+    @Inject
+    CloudAPI.Factory cloudAPIFactory;
+
+    private void validateKMSProviderConfigFormData(ObjectNode formData, String keyProvider) {
+      if(keyProvider.toLowerCase().equals(AWS_KMS_PROVIDER)
+          && (formData.get("AWS_ACCESS_KEY_ID")!=null
+          || formData.get("AWS_SECRET_ACCESS_KEY")!=null)) {
+        CloudAPI cloudAPI = cloudAPIFactory.get(AWS_KMS_PROVIDER);
+        Map<String, String> config = new HashMap<String,String>();
+        config.put("AWS_ACCESS_KEY_ID", formData.get("AWS_ACCESS_KEY_ID").textValue());
+        config.put("AWS_SECRET_ACCESS_KEY", formData.get("AWS_SECRET_ACCESS_KEY")
+            .textValue());
+        if (cloudAPI != null && !cloudAPI.isValidCreds(config, formData.get("AWS_REGION")
+            .textValue())) {
+          throw new RuntimeException("Invalid AWS Credentials.");
+        }
+      }
+      if (keyProvider.equals(SMARTKEY_KMS_PROVIDER)) {
+        Set<String> apiUrl = new HashSet<String>();
+        apiUrl.add("api.amer.smartkey.io");
```

Comment:
Make this set a final class field like: `final Set<String> apiUrl = new HashSet<>(Arrays.asList("api.amer.smartkey.io", "api.eu.smartkey.io", "api.uk.smartkey.io"));` instead of instantiating it each time this method runs.

Reply:
Instead of just verifying that the format of provided creds are valid, you can also actually verify that the credentials are valid by running something like SmartKeyEARService::retrieveSessionAuthorization and checking the output.

Reply:
Sure. Thanks!

Reply:
Would like to merge API error handling change  PR https://github.com/yugabyte/yugabyte-db/pull/8199 before merging this one. 

Will address all the review comments together. This is  OLD PR so may have to create new one. Will update shortly. 

Thanks!